integreat_cms/cms/migrations/0052_alter_role_name.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 3.2.16 on 2022-11-18 13:36
+
+from django.core.management.sql import emit_post_migrate_signal
+from django.db import migrations, models
+from ..constants import roles
+
+# pylint: disable=unused-argument
+def update_roles(apps, schema_editor):
+    """
+    Update permissions for service and management group
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+    Role = apps.get_model("cms", "Role")
+
+    # Emit post-migrate signal to make sure the Permission objects are created before they can be assigned
+    emit_post_migrate_signal(2, False, "default")
+
+    # Clear and update permissions according to new constants
+    for role_name in dict(roles.CHOICES):
+        group, _ = Group.objects.get_or_create(name=role_name)
+        # Clear permissions
+        group.permissions.clear()
+        # Set permissions
+        group.permissions.add(
+            *Permission.objects.filter(codename__in=roles.PERMISSIONS[role_name])
+        )
+
+    # Create the new role Observer
+    observer, _ = Group.objects.get_or_create(name="OBSERVER")
+    Role.objects.get_or_create(name="OBSERVER", group=observer)
```

Comment:
```suggestion
```
The group is already created in the loop, right?
If not, the permissions would be missing.

Reply:
yes, the group is created in the loop before.

Reply:
Yes, the groop is created in the loop.